### PR TITLE
Upstream a few changes we made internally

### PR DESCRIPTION
**Pull request by Y--**
(see comments inline)

### CODE REVIEW

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -19,7 +19,7 @@ endif()
 # ---------------------------------------------------------------------------
 # Ccache
 
-if(EMSCRIPTEN)
+if(EMSCRIPTEN AND EXISTS "$ENV{EMSDK}/ccache/git-emscripten_64bit/bin")
```
Comment (Y--) [inline lib/CMakeLists.txt:22]: This allows compilation on ARM64 plaforms

File: packages/duckdb-wasm/bundle.mjs [hunk 2]
```diff
@@ -175,7 +175,7 @@ fs.copyFile(path.resolve(src, 'bindings', 'duckdb-coi.wasm'), path.resolve(dist,
         bundle: true,
         minify: true,
         sourcemap: is_debug ? 'inline' : true,
-        external: EXTERNALS_WEBWORKER,
+        external: [ EXTERNALS_WEBWORKER, 'module' ],
```
Comment (Y--) [inline packages/duckdb-wasm/bundle.mjs:178]: This is a an initial fix for supporting recent emscripten

File: lib/include/duckdb/web/arrow_stream_buffer.h [hunk 3]
```diff
@@ -85,3 +88,5 @@ struct BufferingArrowIPCStreamDecoder : public arrow::ipc::StreamDecoder {
 
 }  // namespace web
 }  // namespace duckdb
+
+#endif // INCLUDE_DUCKDB_WEB_ARROW_STREAM_BUFFER_H_
```
Comment (ankoh) [inline lib/include/duckdb/web/arrow_stream_buffer.h:92]: Just need a space here for clang-format

**Review by ankoh — CHANGES_REQUESTED** [review]

File: lib/include/duckdb/web/arrow_stream_buffer.h (hunk 3 above)
Comment (Y--) [reply lib/include/duckdb/web/arrow_stream_buffer.h]: Fixed, thanks :-)

**Pull request merged**